src/ImmutablePropTypes.js

```diff
@@ -50,7 +50,7 @@ function getPropType(propValue) {
 function createChainableTypeChecker(validate) {
   function checkType(isRequired, props, propName, componentName, location) {
     componentName = componentName || ANONYMOUS;
-    if (props[propName] == null) {
+    if (props[propName] === null) {
```

Comment:
A clearer answer would be:

```javascript
if (typeof props[propName] === 'undefined' || props[propName] === null) {
```

I cannot speak for others, but I would not have expected `undefined` to pass that test (which is the exact reason that `==` is discouraged).